homeassistant/components/lmstudio/entity.py

```diff
@@ -0,0 +1,243 @@
+"""Base entity for LM Studio integration."""
+
+from __future__ import annotations
+
+from collections.abc import AsyncGenerator, Callable
+import json
+import logging
+from typing import Any
+
+import openai
```

Comment:
Why are you not using the LM Studio native API https://lmstudio.ai/docs/app/api/endpoints/rest